src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/AddFeaturesWithContingentValues/AddFeaturesWithContingentValues.xaml.cs

```diff
@@ -0,0 +1,407 @@
+
+            try
+            {
+                // Instantiate a dictionary containing all possible values for a field in the context of the contingent field groups it participates in.
+                ContingentValuesResult contingentValuesResult = _geodatabaseFeatureTable.GetContingentValues(_newFeature, field);
+
+                // Loop through the contingent values.
+                foreach (ContingentValue contingentValue in contingentValuesResult.ContingentValuesByFieldGroup[fieldGroupName])
+                {
+                    // Contingent coded values are contingent values defined from a coded value domain.
+                    // There are often multiple results returned by the ContingentValuesResult.
+                    if (contingentValue is ContingentCodedValue contingentCodedValue)
+                    {
+                        contingentValuesNamesList.Add(contingentCodedValue.CodedValue.Name);
+                    }
+                    else if (contingentValue is ContingentRangeValue contingentRangeValue)
+                    {
+                        contingentValuesNamesList.Add(contingentRangeValue.MinValue.ToString());
+                        contingentValuesNamesList.Add(contingentRangeValue.MaxValue.ToString());
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                MessageBox.Show(e.Message, "Error");
+            }
+
+            return contingentValuesNamesList;
+        }
+
+        private bool ValidateContingentValues(out List<string> fieldGroupNames, out int numberOfViolations)
+        {
+            fieldGroupNames = new List<string>();
+            numberOfViolations = 0;
+
+            if (_geodatabaseFeatureTable.ContingentValuesDefinition.LoadStatus != Esri.ArcGISRuntime.LoadStatus.Loaded || _newFeature == null) return false;
+
+            IReadOnlyList<ContingencyConstraintViolation> contingencyConstraintViolations = _geodatabaseFeatureTable.ValidateContingencyConstraints(_newFeature);
+            numberOfViolations = contingencyConstraintViolations.Count;
+
+            // If the number of contingency constraint violations is zero the attribute map satisfies all contingencies.
+            if (numberOfViolations.Equals(0))
+            {
+                return true;
+            }
+            else
+            {
+                foreach (ContingencyConstraintViolation violation in contingencyConstraintViolations)
+                {
+                    fieldGroupNames.Add(violation.FieldGroup.Name);
+                }
+            }
+
+            return false;
+        }
+
+        private void CreateNewNest()
+        {
+            // Once the attribute map is filled and validated, save the feature to the geodatabase feature table.
+            _ = _geodatabaseFeatureTable.UpdateFeatureAsync(_newFeature);
```

Comment:
We should await this. There is technically the potential here to query and buffer before the new feature has been updated in the feature table.